qutebrowser/browser/hints.py

```diff
@@ -921,6 +921,10 @@ def fire(self, keystr, force=False):
                           immediately=True)
             return
         if self._context.target in elem_handlers:
+            # Set the pre-jump mark ', so we can jump back here after following
+            tabbed_browser = objreg.get('tabbed-browser', scope='window',
+                                        window=self._win_id)
+            tabbed_browser.set_mark("'")
```

Comment:
When you do this here, the mark will also be set for e.g. downloading via hints.

I think it should be set in the `_click` function instead, and probably only if `context.target` is `Target.normal` or `Target.current`.